numpy/ma/tests/test_core.py

```diff
@@ -1388,6 +1388,21 @@ def test_flat(self):
         assert_equal(b01.data, array([[1., 0.]]))
         assert_equal(b01.mask, array([[False, False]]))
 
+    def test_assign_dtype(self):
+        #check that the mask's dtype is updated when dtype is changed
```

Comment:
This test routine doesn't seem to do what the comment promises. Should there be a few assert's that check the dtype of the mask?